tests/perf/test_accuracy.py

```diff
@@ -2,142 +2,255 @@
+                    # Skip images where gt classes and model classes do not match
+                    continue
+                explanation = self.explainer(
+                    image_np,
+                    targets=intersected_targets,
+                    label_names=self.dataset_label_list,
+                    colormap=False,
+                    **black_box_kwargs,
+                )
+                images.append(image_np)
+                explanations.append(explanation)
+                dataset_gt_bboxes.append(gt_bbox_dict)
+
+            # Write per-batch statistics to track failures
+            explained_images += len(explanations)
+            record = {"range": f"{lrange}-{rrange}"}
+            record.update(self.get_scores_times(explanations, images, dataset_gt_bboxes, start_time))
+            records.append(record)
+
+            df = pd.DataFrame([record]).round(3)
+            df.to_csv(self.data_metric_path / f"accuracy_{self.dataset_name}.csv", mode="a", header=False, index=False)
+
+        experiment_time = time() - experiment_start_time
+        mean_scores_dict = {"explained_images": explained_images, "overall_time": experiment_time}
+        mean_scores_dict.update(
+            {
+                key: np.mean([record[key] for record in records if key in record])
+                for key in records[0].keys()
+                if key != "range"
+            }
+        )
+        df = pd.DataFrame([mean_scores_dict]).round(3)
+        df.to_csv(self.data_metric_path / f"mean_accuracy_{self.dataset_name}.csv", index=False)
 
-            explanation = self.explainer(image_np, targets=targets, label_names=VOC_NAMES, colormap=False)
+    def get_scores_times(
+        self,
+        explanations: list[Explanation],
+        images: list[np.ndarray],
+        dataset_gt_bboxes: Dict[str, List[Tuple[int, int, int, int]]],
+        start_time: float,
+    ):
+        score = {}
+        if len(explanations) == 0:
+            return score
 
-            images.append(image_np)
-            explanations.append(explanation)
-            dataset_gt_bboxes.append({key: value for key, value in gt_bbox_dict.items() if key in targets})
+        def evaluate_and_time(metric_name, evaluation_func, *args, **kwargs):
+            previous_time = time()
+            score.update(evaluation_func(*args, **kwargs))
+            score[f"{metric_name}_time"] = time() - previous_time
 
-        pointing_game = self.pointing_game.evaluate(explanations, dataset_gt_bboxes)
-        auc = self.auc.evaluate(explanations, images, steps=10)
-        adcc = self.adcc.evaluate(explanations, images)
+        score["explain_time"] = time() - start_time
+        evaluate_and_time("pointing_game", self.pointing_game.evaluate, explanations, dataset_gt_bboxes)
+        evaluate_and_time("auc", self.auc.evaluate, explanations, images, steps=30)
+        evaluate_and_time("adcc", self.adcc.evaluate, explanations, images)
```

Comment:
what if it is required to measure only one metric?

Reply:
For regression tests, we gather all the metrics, so if some custom set is required, it can be done by commenting out the line. I don't think that on this stage we need to support update the set of metrics through CLI